app/models/host.rb

```diff
@@ -35,12 +35,8 @@ class Host < ApplicationRecord
   has_many :assigned_internal_profiles, -> { external(false) },
            through: :policies, source: :profiles
 
-  scope :with_benchmark, lambda { |profile = nil|
-    profile ||= RequestStore.store['scoped_search_context_profiles']
-
-    left_outer_joins(test_result_profiles: :benchmark).where(
-      profiles: { id: profile.pluck(:id) }
-    )
```

Comment:
@skateman The issue with the sorting of the ssg version was caused by `profile` being nil. I tried to figure out how to set the profile correctly but then after looking at it for a little while, I was wondering if the last part of the query was necessary? It seems like `left_outer_joins(test_result_profiles: :benchmark)` might be sufficient? Or maybe I'm not understanding what `test_result_profiles` is exactly?
It's hard to test because the ssg versions are showing up as "Not available" on this table.. I asked Egor if he noticed the same thing when he was testing

Reply:
@marleystipich2 your approach would only work if one host could be only assigned to a single profile and so have one test result and one benchmark. However, we need the profile(s) as a parameter to distinguish which policy we're underneath at the moment. If you delete the `profile` variable, it will also break the SSG sorting on a report's page.

So in order to have this working, you need to set the `profile` variable that's coming from the `RequestStore`. 

I'd suggest you to [look](https://github.com/RedHatInsights/compliance-backend/blob/master/app/models/concerns/host_searching.rb#L82) where and how the `RequestStore.store['scoped_search_context_profiles']` is being set when opening a report's page.Then you should figure out why this is not being called when opening a policy/systems page.

Reply:
@skateman Okay, right I figured that that was the reason that checking the profile was necessary. This make sense! 
I'll look into why `RequestStore.store['scoped_search_context_profiles']` isn't being set correctly when you open the policy/systems page. I was looking at the method where it's being set and noticed that it wasn't being called when you're at this page. Thanks for the help!!